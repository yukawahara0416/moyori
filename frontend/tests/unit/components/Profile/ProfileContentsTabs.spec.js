import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Component from '@/components/Profile/ProfileContentsTabs.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper
let store
let tab

beforeEach(() => {
  tab = {
    getters: {
      profileTab: () => 'test'
    }
  }

  store = new Vuex.Store({
    modules: {
      tab
    }
  })

  wrapper = shallowMount(Component, {
    localVue,
    store
  })
})

describe('getters', () => {
  it('profileTab', () => {
    expect(wrapper.vm.profileTab).toEqual(tab.getters.profileTab())
  })
})

describe('computed', () => {
  it('childTabs/get', () => {
    expect(wrapper.vm.childTabs).toEqual(tab.getters.profileTab())
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
