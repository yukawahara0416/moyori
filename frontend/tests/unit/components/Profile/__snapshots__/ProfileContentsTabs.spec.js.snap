// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-tabs-stub
  activeclass=""
  fixedtabs="true"
  mobilebreakpoint="1264"
  nexticon="$next"
  previcon="$prev"
  slidersize="2"
  value="tab-1"
>
  <v-tab-stub
    class="primary--text"
    href="#tab-1"
    ripple="true"
  >
    <v-icon-stub>
      mdi-map-marker-outline
    </v-icon-stub>
  </v-tab-stub>
   
  <v-tab-stub
    class="primary--text"
    href="#tab-2"
    ripple="true"
  >
    <v-icon-stub>
      mdi-wifi
    </v-icon-stub>
     
    <v-icon-stub>
      mdi-power-plug
    </v-icon-stub>
  </v-tab-stub>
   
  <v-tab-stub
    class="primary--text"
    href="#tab-3"
    ripple="true"
  >
    <v-icon-stub>
      mdi-message-outline
    </v-icon-stub>
  </v-tab-stub>
   
  <v-tab-stub
    class="primary--text"
    href="#tab-4"
    ripple="true"
  >
    <v-icon-stub>
      mdi-heart-outline
    </v-icon-stub>
  </v-tab-stub>
</v-tabs-stub>
`;
